File: patch_search.py
import os

filepath = 'server/routes/Routes.js'
with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

bad_pattern = "const regex = new RegExp(search.split(' ').join('|'), 'i');"
good_pattern = r"""const escapedSearch = search.trim().replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
      const regex = new RegExp(escapedSearch, 'i');"""

content = content.replace(bad_pattern, good_pattern)

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(content)

print("Replaced all bad regexes in Routes.js")

File: patch_search.js
const fs = require('fs');
const content = fs.readFileSync('server/routes/Routes.js', 'utf8');

const regexSplit = /const regex = new RegExp\(search\.split\(' '\)\.join\('\|'\), 'i'\);/g;
const regexTrim = /const regex = new RegExp\(search\.trim\(\), 'i'\);/g;
const replacement = 'const escapedSearch = search.trim().replace(/[.*+?^${}()|[\\]\\\\]/g, \'\\\\$&\');\n      const regex = new RegExp(escapedSearch, \'i\');';

let newContent = content.replace(regexSplit, replacement);
newContent = newContent.replace(regexTrim, replacement);

fs.writeFileSync('server/routes/Routes.js', newContent);
console.log('Successfully replaced regex creations in Routes.js');
